src/server/Controllers/StoresController.cs

```diff
@@ -149,59 +144,42 @@ public async Task<IActionResult> GetStoreCatchAll(string storeId, string catchAl
                 return Redirect($"{referer}/{storeId}/{key}");
             }
 
+            // Requesting GetValue only from the last root hash onchain ensures that only
+            // nodes that have the latest state will respond to the request
+            // This helps prevent a mismatch between the state of the store and
+            // the data when pulled across decentralized nodes
+            var lastRootHash = await _gatewayService.GetLastRoot(storeId, cancellationToken);
+            if (lastRootHash is null)
+            {
+                return NotFound();
```

Comment:
Just for readability, i wonder if we should make a custom `StoreNotFound()`

Reply:
That's the AspNet way of returning 404. Are you thinking `StoreNotFound` would do anything other than that?